src/lib/boavizta/index.ts

```diff
@@ -164,219 +55,178 @@ export class BoaviztaCpuOutputModel
-        )
-      ) {
-        throw new UnsupportedValueError(
-          this.errorBuilder({
-            message: `Invalid 'instance-type' parameter: '${
-              staticParamsCast['instance-type']
-            }'. Valid values are : ${this.instanceTypes[provider].join(', ')}`,
-          })
-        );
-      }
-    }
+    return this.sharedParams;
   }
 
-  async validateProvider(staticParamsCast: object) {
-    if (!('provider' in staticParamsCast)) {
+  /**
+   * Fetches data from the Boavizta API for the Cloud model.
+   */
+  protected async fetchData(usageData: object | undefined): Promise<object> {
+    if (this.sharedParams === undefined) {
       throw new InputValidationError(
         this.errorBuilder({
-          message: "Missing 'provider' parameter from configuration",
+          message: 'Missing configuration parameters',
         })
       );
-    } else {
-      const supportedProviders = await this.supportedProvidersList();
-
-      if (!supportedProviders.includes(staticParamsCast.provider as string)) {
-        throw new InputValidationError(
-          this.errorBuilder({
-            message: `Invalid 'provider' parameter '${
-              staticParamsCast.provider
-            }'. Valid values are ${supportedProviders.join(', ')}`,
-          })
-        );
-      }
     }
-  }
 
-  async supportedInstancesList(provider: string) {
-    const instances = await axios.get(
-      `https://api.boavizta.org/v1/cloud/instance/all_instances?provider=${provider}`
-    );
+    const dataCast: object = Object.assign({}, this.sharedParams, {
+      usage: usageData,
+    });
 
-    return instances.data;
+    const response = await this.boaviztaAPI.fetchCloudInstanceData(
+      dataCast,
+      this.verbose
+    );
+    return this.formatResponse(response);
   }
 
-  async supportedProvidersList(): Promise<string[]> {
-    const providers = await axios.get(
-      'https://api.boavizta.org/v1/cloud/instance/all_providers'
-    );
+  /**
+   * Validates static parameters for the Cloud model using Zod schema.
+   */
+  private validateStaticParams(staticParams: ModelParams) {
+    const schema = z
+      .object({
+        provider: z.string(),
+        'instance-type': z.string(),
+        verbose: z.boolean().optional(),
+        'expected-lifespan': z.number().optional(),
+      })
+      .refine(allDefined, {message: 'All parameters are required.'});
 
-    return Object.values(providers.data);
+    return validate<z.infer<typeof schema>>(schema, staticParams);
   }
 
-  async fetchData(usageData: object | undefined): Promise<object> {
-    if (this.sharedParams === undefined) {
+  /**
+   * Validates the provider parameter for the Cloud model.
+   */
+  private async validateProvider(staticParamsCast: ModelParams) {
+    const supportedProviders =
+      await this.boaviztaAPI.getSupportedProvidersList();
+
+    if (!supportedProviders.includes(staticParamsCast.provider as string)) {
```

Comment:
if possible please reduce usage of explicit assignment

Reply:
done